src/platform/Linux/bluez/ChipDeviceScanner.h

```diff
@@ -96,11 +110,9 @@ class ChipDeviceScanner
     ChipDeviceScannerDelegate * mDelegate = nullptr;
     gulong mObjectAddedSignal             = 0;
     gulong mInterfaceChangedSignal        = 0;
-    bool mIsInitialized                   = false;
-    bool mIsScanning                      = false;
-    bool mIsStopping                      = false;
+    ChipDeviceScannerState mScannerState  = ChipDeviceScannerState::SCANNER_UNINITIALIZED;
     /// Used to track if timer has already expired and doesn't need to be canceled.
-    bool mTimerExpired = false;
+    ScannerTimerState mTimerState = ScannerTimerState::TIMER_CANCELED;
```

Comment:
Calling `CancelTimer` when the timer is not started is not a big deal, so keeping the timer state might not really be that useful, but it shouldn't harm either :)

Reply:
Yes, I just kept the original logic here